util/src/main/java/tc/oc/pgm/util/tablist/TabEntry.java

```diff
@@ -54,6 +56,13 @@
    */
   BaseComponent[] getContent(TabView view);
 
+  /**
+   * Content to show in the entry
+   *
+   * @return
+   */
+  Component getComponent(TabView view);
+
```

Comment:
Having two methods to get the component is ambiguous and generally a bad idea. 
It was changed to `BaseComponent[]` for performance reasons when rendering complex components (like playerrnames) to avoid extra layers of nesting on serialization of the component.

If you wish, you can have this as an abstract method in `SimpleTabEntry`, and implement the `BaseComponent[]` version there, that by default will convert the `Component` to `BaseComponent[]` (which is what you're doing either way) but it should not be part of the interface.

With the current changes, any external tab entry implementation that implements `BaseComponent[]` but not the `Component` version will be either different for 1.7 players than 1.8 players, or it will simply break, depending on if it extends pgm's tab entry or not.

Reply:
Every single implementation of getContent for TabEntry in this repository makes an adventure component and then calls TextTranslations.toBaseComponentArray to turn it into a BaseComponent. I added the Component version so that we could call the adventure set playerlistheaderfooter method, which actually cuts down on serialization/deserialization for that. Also, at least for the pgm implementations, this won't affect performance for the rest of the tab list.

Reply:
I've removed the original getContent method, which resolves the ambiguity. This will still break any external tab entry that implements the BaseComponent version.

Reply:
Take this as an example:
https://github.com/bolt-rip/ingame/blob/master/src/main/java/rip/bolt/ingame/utils/RankedTeamTabEntry.java

It works fast due to being able to edit the component by simply using array manipulation, works fast and works well. This isn't the only place it's used

Reply:
So then the way I see it is it's a tradeoff, is it worth making a breaking API change for this.

Switching the tab list to using Adventure Components reduces serialization and deserialization of components.

As an aside, the linked RankedTeamTabEntry class would still be faster if it used an Adventure Component; in its current state the TeamTabEntry superclass makes an Adventure Component then converts it to a BaseComponent Array then the RankedTeamTabEntry class modifies that array, and then that is serialized by SportPaper to a gson string. It would be faster to override the Adventure Component version from this pull request which would then be directly converted to a gson string. Although you do know better than me what else this would break.

I think it's worth it to get rid of BaseComponents from the Tab list and increment the PGM version number for a breaking API change.

@Electroid  thoughts?